diff --git a/rs/private/rustc_repository.bzl b/rs/private/rustc_repository.bzl
--- a/rs/private/rustc_repository.bzl
+++ b/rs/private/rustc_repository.bzl
@@ -60,6 +60,6 @@ _LINUX_ZLIB = {
     "x86_64": struct(
         libdir = "usr/lib/x86_64-linux-gnu",
         sha256 = "7074b6a2f6367a10d280c00a1cb02e74277709180bab4f2491a2f355ab2d6c20",
-        url = "https://archive.ubuntu.com/ubuntu/pool/main/z/zlib/zlib1g_1.3.dfsg-3.1ubuntu2.1_amd64.deb",
+        url = "https://launchpadlibrarian.net/742985810/zlib1g_1.3.dfsg-3.1ubuntu2.1_amd64.deb",
     ),
 }
